climada/hazard/tc_tracks_synth.py

```diff
@@ -726,37 +748,60 @@ def _apply_decay_coeffs(track, v_rel, p_rel, land_geom, s_rel):
+            if np.any(p_decay > 1):
+                LOGGER.warning('Landfall decay would decrease pressure for '
+                               'track id %s. Decay not applied, please check.',
+                               track.sid)
+                p_decay[p_decay < 1] = track.central_pressure[sea_land:land_sea][p_decay < 1] / p_landfall
+            track.central_pressure[sea_land:land_sea] = p_landfall * p_decay
 
         v_decay = _decay_v_function(v_rel[ss_scale_idx],
                                     track.dist_since_lf[sea_land:land_sea].values)
-        # dont applay decay if it would increas wind speeds
-        v_decay[v_decay > 1] = (track.max_sustained_wind[sea_land:land_sea][v_decay > 1]
-                                / v_landfall)
+        # dont apply decay if it would increase wind speeds
+        if np.any(v_decay > 1):
+            # should not happen unless v_rel is negative
+            LOGGER.warning('Landfall decay would increase wind speed for '
+                           'track id %s. Decay not applied, please check.',
+                           track.sid)
+            v_decay[v_decay > 1] = (track.max_sustained_wind[sea_land:land_sea][v_decay > 1]
+                                    / v_landfall)
         track.max_sustained_wind[sea_land:land_sea] = v_landfall * v_decay
 
-        # correct values of sea between two landfalls
-        if land_sea < track.time.size and idx + 1 < sea_land_idx.size:
+        # correct values of sea after a landfall (until next landfall, if any)
+        if land_sea < track.time.size:
+            if idx + 1 < sea_land_idx.size:
+                # if there is a next landfall, correct until last point before
+                # reaching land again
+                end_cor = sea_land_idx[idx + 1]
+            else:
+                # if there is no further landfall, correct until the end of
+                # the track
+                end_cor = track.time.size
             rndn = 0.1 * float(np.abs(np.random.normal(size=1) * 5) + 6)
             r_diff = track.central_pressure[land_sea].values - \
                      track.central_pressure[land_sea - 1].values + rndn
-            track.central_pressure[land_sea:sea_land_idx[idx + 1]] += - r_diff
+            track.central_pressure[land_sea:end_cor] += - r_diff
 
             rndn = rndn * 10  # mean value 10
             r_diff = track.max_sustained_wind[land_sea].values - \
                      track.max_sustained_wind[land_sea - 1].values - rndn
-            track.max_sustained_wind[land_sea:sea_land_idx[idx + 1]] += - r_diff
+            track.max_sustained_wind[land_sea:end_cor] += - r_diff
+
+        # correct limits
+        np.warnings.filterwarnings('ignore')
+        cor_p = track.central_pressure.values > track.environmental_pressure.values
+        track.central_pressure[cor_p] = track.environmental_pressure[cor_p]
+        track.max_sustained_wind[track.max_sustained_wind < 0] = 0
```

Comment:
This code block was moved within the loop through landfalls, so that a non-first landfall already has sensible values to deal with. It helps solving the issue of no pressure correction after landfall.